tests/test_config_entries.py

```diff
@@ -1577,3 +1577,198 @@ async def async_step_import(self, user_input):
         assert len(entries) == 1
         assert entries[0].state == config_entries.ENTRY_STATE_LOADED
         assert entries[0].data == {"value": "updated"}
+
+
+async def test_flow_with_default_discovery(hass, manager):
```

Comment:
Could you parametrize this test so you test each source.

Reply:
This test is a subset of `test_finish_flow_removed_default_discovery_unique_id`.